velox/dwio/common/tests/TestBufferedInput.cpp

```diff
@@ -31,3 +35,67 @@ TEST(TestBufferedInput, ZeroLengthStream) {
   EXPECT_FALSE(ret->Next(&buf, &size));
   EXPECT_EQ(size, 0);
 }
+
+TEST(TestBufferedInput, ParallelLoad) {
+  const int32_t kOneKB = 1 * 1024;
+  const int32_t threads = 4;
+  FLAGS_load_parallel = true;
+
+  std::shared_ptr<IoStatistics> ioStats = std::make_shared<IoStatistics>();
+  std::unique_ptr<folly::Executor> executor =
+      std::make_unique<folly::IOThreadPoolExecutor>(threads);
+  int32_t loadQuantum = kOneKB;
+
+  std::string fileData;
+  {
+    InMemoryWriteFile writeFile(&fileData);
+    writeFile.append(std::string(5, 'a'));
+    writeFile.append(std::string(kOneKB, 'b'));
+    writeFile.append(std::string(kOneKB, 'c'));
+    writeFile.append(std::string(5, 'd'));
+  }
+  auto readFile = std::make_shared<facebook::velox::InMemoryReadFile>(fileData);
+  auto pool = facebook::velox::memory::getDefaultMemoryPool();
+  BufferedInput input(
+      readFile,
+      *pool,
+      MetricsLog::voidLog(),
+      ioStats,
+      executor.get(),
+      loadQuantum,
+      1);
+
+  std::vector<Region> regions = {
+      {0, kOneKB},
+      {kOneKB, 10},
+      {kOneKB + 20, readFile->size() - (kOneKB + 20)}};
+  std::vector<std::unique_ptr<SeekableInputStream>> streams;
+  for (auto& region : regions) {
+    streams.push_back(input.enqueue(region));
+  }
+
+  input.load(LogType::TEST);
+
+  for (size_t i = 0; i < regions.size(); i++) {
+    const auto& region = regions[i];
+    const auto& stream = streams[i];
+
+    auto buffer =
+        std::make_shared<dwio::common::DataBuffer<char>>(*pool, region.length);
+    stream->readFully(buffer->data(), region.length);
+
+    // verify data correctness.
```

Comment:
Make sure comments are full sentences, e.g. start with a capital letter and end with a period.